base/essentials.jl

```diff
@@ -439,6 +439,19 @@ julia> reinterpret(Float32, UInt32[1 2 3 4 5])
 1×5 reinterpret(Float32, ::Matrix{UInt32}):
  1.0f-45  3.0f-45  4.0f-45  6.0f-45  7.0f-45
 ```
+
+For lazy array containers, the `reinterpret` works on an element-wise sense, so the memory
+might not reflect the actual memory storage:
+
+```jldoctest
+julia> reinterpret(Int, CartesianIndices((2, 2))) # `CartesianIndices` doesn't allocates memory
```

Comment:
```suggestion
`reinterpret` does not rely on having a dense block of memory. Hence, it also supports "lazy" arrays: 

```jldoctest
julia> reinterpret(Int, CartesianIndices((2, 2))) # `CartesianIndices` does not have a buffer holding array values
```